x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/aggregation/CountAggregatorFunction.java

```diff
@@ -57,24 +58,31 @@ public static CountAggregatorFunction create(List<Integer> inputChannels) {
     private CountAggregatorFunction(List<Integer> channels, LongState state) {
         this.channels = channels;
         this.state = state;
+        this.countAll = channels.isEmpty();
     }
 
     @Override
     public int intermediateBlockCount() {
         return intermediateStateDesc().size();
     }
 
+    private int blockIndex() {
+        return countAll ? 0 : channels.get(0);
```

Comment:
@nik9000 not sure if this is the best way to count things in a block but it seems to be working.